int32 isMoving
int32 isPuting
int32 singleSortOK
string OverLoad